### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by daun**
**Description**

- Unit tests only
- For this plugin, we don't really need end-to-end tests
  - Unit-testing the helpers
  - Unit-testing the plugin to make sure it calls the helpers
  - That's about as good as messing with the whole end-to-end setup and much simpler :)

**Checks**

- [x] The PR is submitted to the `master` branch
- [x] The code was linted before pushing (`npm run lint`)
- [x] All tests are passing (`npm run test`)
- [x] New or updated tests are included
- [ ] ~~The documentation was updated as required~~

### CODE REVIEW

**Comment by daun** [comment]
@hirasso Ready for review :)

**Comment by daun** [comment]
@hirasso Just realized this looks a bit chaotic from the diff view — I fell for the temptation of some refactoring 🤠 but most of it is related to making things testable or applying autoformatting.

File: tests/config/vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		environmentOptions: {
+			jsdom: {
+				console: true,
+				resources: 'usable'
+			}
+		},
+		include: ['tests/unit/**/*.test.ts'],
+		setupFiles: ['tests/config/vitest.setup.ts'],
+		testTimeout: 1000,
+		browser: browser ? {
```
Comment (hirasso) [inline tests/config/vitest.config.ts]: Do we want to keep the `browser` setting here for future reference? If not, I'd remove it

File: tests/unit/mergeHeadContents.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// FILEPATH: /Users/philipp/Projects/@swup/head-plugin/tests/unit/mergeHeadContents.test.ts
```
Comment (hirasso) [inline tests/unit/mergeHeadContents.test.ts]: What's this comment doing?

**Review by hirasso — COMMENTED** [review]
This is all all well over my head as I haven't ever looked into the source code of head plugin and haven't got the time to do so right now – that's why my comments are only small nitpicks. Sorry for not being of more help here, but tests are always better then no tests right? 😇

**Review by hirasso — DISMISSED** [review]

File: tests/unit/mergeHeadContents.test.ts (hunk 2 above)
Comment (daun) [reply tests/unit/mergeHeadContents.test.ts]: Nothing, of course 🥴 Looks automatically generated, so I assumed it had to do with some obscure vitest feature, but turns it it's entirely optional.

File: tests/config/vitest.config.ts (hunk 1 above)
Comment (daun) [reply tests/config/vitest.config.ts]: Right, it's easy to restore in case we ever need to.

**Comment by hirasso** [comment]
@daun did you test the latest @swup/cli `swup package:format` with this, formatting the tests and readme.md additionally to the source code? 🙂

**Comment by daun** [comment]
@hirasso I think so?

**Comment by daun** [comment]
@hirasso Looks like that's what happened, but I don't remember 🤠

**Review by hirasso — APPROVED** [review]
🧪

**Pull request merged**